src/linquantarrays.jl

```diff
@@ -10,84 +13,164 @@ Base.size(QA::LinQuantArray) = size(QA.A)
 Base.getindex(QA::LinQuantArray,i...) = getindex(QA.A,i...)
 Base.eltype(Q::LinQuantArray{T,N}) where {T,N} = T
 
-"""Quantise an array linearly into a LinQuantArray."""
-function LinQuantization(::Type{T},A::AbstractArray) where {T<:Unsigned}
+"""
+
+        LinQuantization(::Type{T}, A::AbstractArray; extrema::Tuple = extrema(A)) where {T<:Integer}
+
+Quantise an array linearly into a LinQuantArray.
+
+# Arguments
+- `T`: the type of the quantised array
+- `A`: the array to quantise
+- `extrema`: the minimum and maximum of the range, defaults to `extrema(A)`.
+
+# Returns
+- a LinQuantArray{T} with the quantised array and the minimum and maximum of the original range.
+"""
+
+function LinQuantization(
+    ::Type{T},
+    A::AbstractArray,
+    extrema::Nothing
```

Comment:
I believe this should be `extrema::Nothing = nothing` so that the default uses the non-clamped version?